cmd/subnetcmd/import_from_api.go

```diff
@@ -0,0 +1,282 @@
+	options := []rpc.Option{}
+
+	ux.Logger.PrintToUser("Getting information from the %s network...", network.String())
+
+	chains, err := client.GetBlockchains(ctx, options...)
+	if err != nil {
+		return err
+	}
+
+	var (
+		blockchainID, vmID ids.ID
+		subnetName         string
+	)
+
+	for _, ch := range chains {
+		// NOTE: This supports only one chain per subnet
+		if ch.SubnetID == subnetID {
+			blockchainID = ch.ID
+			vmID = ch.VMID
+			subnetName = ch.Name
+			break
+		}
+	}
+
+	if blockchainID == ids.Empty || vmID == ids.Empty {
+		return fmt.Errorf("subnet ID %s not found on this network", subnetIDstr)
+	}
+
+	ux.Logger.PrintToUser("Retrieved information. BlockchainID: %s, Name: %s, VMID: %s",
+		blockchainID.String(),
+		subnetName,
+		vmID.String(),
+	)
+	// TODO: it's probably possible to deploy VMs with the same name on a public network
+	// In this case, an import could clash because the tool supports unique names only
+
+	genBytes, err := os.ReadFile(genesisFilePath)
+	if err != nil {
+		return err
+	}
+
+	if err = app.WriteGenesisFile(subnetName, genBytes); err != nil {
+		return err
+	}
+
+	vmType := getVMFromFlag()
+	if vmType == "" {
+		subnetTypeStr, err := app.Prompt.CaptureList(
+			"What's this VM's type?",
+			[]string{models.SubnetEvm, models.SpacesVM, models.CustomVM},
+		)
+		if err != nil {
+			return err
+		}
+		vmType = models.VMTypeFromString(subnetTypeStr)
+	}
+
+	sc := &models.Sidecar{
+		Name:            subnetName,
+		VM:              vmType,
+		ImportedFromAPM: false,
+		Networks: map[string]models.NetworkData{
+			network.String(): {
+				SubnetID:     subnetID,
+				BlockchainID: blockchainID,
+			},
+		},
+		Subnet:  subnetName,
+		Version: constants.SidecarVersion,
+	}
+
+	var versions []string
+	switch vmType {
+	case models.SubnetEvm:
+		versions, err = app.Downloader.GetAllReleasesForRepo(constants.AvaLabsOrg, constants.SubnetEVMRepoName)
+		if err != nil {
+			return err
+		}
+		sc.VMVersion, err = app.Prompt.CaptureList("Pick the version for this VM", versions)
+	case models.SpacesVM:
+		versions, err = app.Downloader.GetAllReleasesForRepo(constants.AvaLabsOrg, constants.SpacesVMRepoName)
+		if err != nil {
+			return err
+		}
+		sc.VMVersion, err = app.Prompt.CaptureList("Pick the version for this VM", versions)
+	case models.CustomVM:
+		return fmt.Errorf("importing custom VMs is not yet implemented, but will be available soon")
+	default:
+		return fmt.Errorf("unexpected VM type: %v", vmType)
+	}
+	if err != nil {
+		return err
+	}
+
+	// hasn't been set in reply
+	if sc.RPCVersion == 0 {
+		sc.RPCVersion, err = vm.GetRPCProtocolVersion(app, vmType, sc.VMVersion)
+		if err != nil {
+			return fmt.Errorf("failed getting RPCVersion for VM type %s with version %s", vmType, sc.VMVersion)
+		}
+	}
+
+	if vmType == models.SubnetEvm {
+		var genesis core.Genesis
+		if err := json.Unmarshal(genBytes, &genesis); err != nil {
+			return err
+		}
+		sc.ChainID = genesis.Config.ChainID.String()
+		/*
```

Comment:
No, it's not encoded in the genesis, we only track it for metamask purposes. Set above and delete here.